cli/command/stack/kubernetes/cli.go

```diff
@@ -57,7 +53,19 @@ func WrapCli(dockerCli command.Cli, opts Options) (*KubeCli, error) {
 			kubeConfig = filepath.Join(homedir.Get(), ".kube/config")
 		}
 	}
-	config, err := kubernetes.NewKubernetesConfig(kubeConfig)
+
+	clientConfig := kubernetes.NewKubernetesConfig(kubeConfig)
+
+	configNamespace, _, err := clientConfig.Namespace()
+	if err != nil {
+		return nil, err
+	}
+	cli.kubeNamespace = configNamespace
+	if opts.Namespace != "default" {
```

Comment:
- What if `opts.Namespace` is an empty string? (also; I'm not a big fan of magic words, such as `"default"` for default values) 
- What if I run `docker stack ls --namespace=default`; will the specified (`"default"`) namespace be ignored? (perhaps `kubernetes.NewOptions()` should use `flags.Changed()` to detect if the flag was set (even if it's with the default value https://github.com/docker/cli/blob/854aad8927e9e61c10be944ced7546f0b7142379/cli/command/stack/kubernetes/cli.go#L31-L41)

Reply:
Running a complete set of (manual) tests with both ```kubectl get stacks``` and ```docker stack ls``` reveals a few discrepancies. I'll sort that out.